controller/payment_handler.go

```diff
@@ -0,0 +1,59 @@
+package controller
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cyneptic/letsgo/internal/core/service"
+	"github.com/labstack/echo/v4"
+)
+
+type PaymentHandlers struct {
+	srv service.PaymentService
+}
+
+func NewPaymentHandler(srv service.PaymentService) *PaymentHandlers {
+	return &PaymentHandlers{
+		srv: srv,
+	}
+}
+
+func RegisterPaymentRoutes(e *echo.Echo, srv service.PaymentService) {
+	handler := NewPaymentHandler(srv)
+	e.GET("/create-payment", handler.CreatePayment)
+	e.POST("/verify-payment", handler.VerifyPayment)
+}
+
+func (p *PaymentHandlers) CreatePayment(e echo.Context) error {
+	reservationId := e.QueryParam("reservation_id")
+	if reservationId == "" {
+		return echo.ErrBadRequest
+	}
```

Comment:
validations should be in a controller/validators/ folder in a relative *_validator.go